Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -1292,9 +1311,17 @@ public void closeSession() {
      * <p>Closes the current session. Should be called by on getting the last actvity onStop() event.
      * </p>
      */
-    private void closeSessionInternal() {
+    void closeSessionInternal() {
         executeClose();
         sessionReferredLink_ = null;
+        trackingController.updateTrackingState(context_); // Update the tracking state for next cold start
+    }
+    
+    /**
+     * Clears all pending requests in the queue
+     */
+    void clearPendingRequests() {
+        requestQueue_.clear();
```

Comment:
If this is called while a request is in progress (say v1/open), it'll still return and set all of the preference helper values after you cleared them in TrackingController, right?

Reply:
In tracking disabled mode, we are not adding any tracking related params to preference helper  when request is responded 
https://github.com/BranchMetrics/android-branch-deep-linking/blob/Tracking-Control/Branch-SDK/src/io/branch/referral/Branch.java#L2877

Reply:
I saw that later. Thanks! Deleting this comment